### ISSUE TITLE
no-cache directive is ignored

### ISSUE THREAD

**Issue by wsargent** [playframework/cachecontrol#35]
From [#32](https://github.com/playframework/cachecontrol/pull/32):

```
It's not possible for `requestContainsNoCache`
to be Some(Validate(msg)) in the presence of
Cache-control: no-cache; only when the Pragma was present.

The request.directives collection was being searched for the
NoCache.type instead of a class of type NoCache.
```

For reference, no-cache directive is defined in https://tools.ietf.org/html/rfc7234#section-5.2.1.4

Previous PR was closed due to lack of legal signoff, so I'll work off this bug without reference to PR.

Note that the tests are changed because the no-cache tests were incorrectly testing for staleness, and there was no explicit test for a no-cache pragma.  In short, if "no-cache" exists you should always validate.

![response caching directive](http://bizcoder.com/Media/Bizcoder/Windows-Live-Writer/Caching_FA94/image_12.png)

**Comment by bpossolo** [comment]
our legal team finally approved the contribution
https://github.com/playframework/cachecontrol/pull/32

**Comment by dwijnand** [comment]
Fixed in #36.
